src/main/kotlin/graphics/scenery/controls/behaviours/VRGrab.kt

```diff
@@ -0,0 +1,139 @@
+package graphics.scenery.controls.behaviours
+
+import graphics.scenery.Node
+import graphics.scenery.Scene
+import graphics.scenery.attribute.spatial.Spatial
+import graphics.scenery.controls.OpenVRHMD
+import graphics.scenery.controls.TrackedDeviceType
+import graphics.scenery.controls.TrackerRole
+import graphics.scenery.utils.extensions.minus
+import graphics.scenery.utils.extensions.plus
+import org.joml.Quaternionf
+import org.joml.Vector3f
+import org.scijava.ui.behaviour.DragBehaviour
+
+
+/**
+ * Grab and Drag nodes with a VR controller.
+ *
+ * @param targets Only nodes in this list may be dragged. They must have a [Grabable] attribute.
+ * @param multiTarget If this is true all targets which collide with [controllerHitbox] will be dragged otherwise only one.
+ *
+ * @author Jan Tiemann
+ */
+open class VRGrab(
+    protected val name: String,
+    protected val controllerHitbox: Node,
+    protected val targets: () -> List<Node>,
+    protected val multiTarget: Boolean = false,
+    protected val onGrab: (() -> Unit)? = null
+) : DragBehaviour {
+
+    val controllerSpatial: Spatial
+
+    init {
+        controllerSpatial = controllerHitbox.spatialOrNull()
+            ?: throw IllegalArgumentException("controller hitbox needs a spatial attribute")
+    }
+
```

Comment:
Should all of these vars be public indeed?

Reply:
they should be protected. There is already a child class who uses this in my microscenery.